test/functional/feature_rbf.py

```diff
@@ -623,5 +642,130 @@ def test_replacement_relay_fee(self):
+        joined_tx_hex = self.create_double_input_self_transfer(parent_utxos, Decimal('0.0001'), sequence=0xffffffff)
+        joined_tx_txid = self.nodes[0].sendrawtransaction(joined_tx_hex)
+
+        # Only a single input needs to signal replaceability (whether explicitly or through inheritance) for the resulting transaction to signal
+        assert_equal(True, self.nodes[0].getmempoolentry(joined_tx_txid)['bip125-replaceable'])
+
+        # Get the `joined_tx` utxo into our wallet so we can spend a chain of descendants from it
+        joined_tx = self.nodes[0].decoderawtransaction(joined_tx_hex)
+        self.wallet.scan_tx(joined_tx)
+        joined_utxo = self.wallet.get_utxo(txid=joined_tx_txid)
+
+        # If we confirm `optin_parent_tx` descendants will no longer inherit signaling
+        hash = self.generateblock(self.nodes[0], output=self.wallet.get_address(), transactions=[optin_parent_tx['txid']])['hash']
+        assert_equal(False, self.nodes[0].getmempoolentry(joined_tx_txid)['bip125-replaceable'])
+
+        # Get `optin_parent_tx` back in our mempool, once again descendants inherit signaling
+        self.nodes[0].invalidateblock(hash)
+        assert_equal(True, self.nodes[0].getmempoolentry(joined_tx_txid)['bip125-replaceable'])
+
+        # Create a chain the size of `MAX_REPLACEMENT_LIMIT` spending `joined_tx` - the BIP125 Rule #5 imposed limit
+        tx = None
+        for _ in range(MAX_REPLACEMENT_LIMIT - 1):
+            tx = self.wallet.send_self_transfer(
+                from_node=self.nodes[0],
+                utxo_to_spend=joined_utxo if tx is None else self.wallet.get_utxo(txid=tx['txid']),  # a straight line of descendants
+                sequence=0xffffffff,
+                fee_rate=Decimal('0.0001'),
+            )
+            assert_equal(True, self.nodes[0].getmempoolentry(tx['txid'])['bip125-replaceable'])  # inherited
+        # Now we have a chain of: `optin_parent_tx`, `joined_tx`, and 99 txs. The last tx in the loop exceeded `MAX_REPLACEMENT_LIMIT`
+
+        # Attempting to replace the opt-in parent transaction will now result in more than `MAX_REPLACEMENT_LIMIT`
+        # conflicting txns being evicted from the mempool. However, it (and all of its descendants) are still signaling replaceability.
+        # We would've expected that once `MAX_REPLACEMENT_LIMIT` is exceeded, the opt-in parent txn stops signaling
+        # replaceability, along with _all_ of its descendants.
+        entry = self.nodes[0].getmempoolentry(optin_parent_tx["txid"])
+        assert_greater_than(entry['descendantcount'], MAX_REPLACEMENT_LIMIT)
+        assert_equal(True, entry['bip125-replaceable'])
+        assert_equal(True, self.nodes[0].getmempoolentry(tx["txid"])['bip125-replaceable'])
+
+        # Case in point, we can't actually replace `optin_parent_tx` once it has `MAX_REPLACEMENT_LIMIT` descendants
+        self.wallet.create_self_transfer(
```

Comment:
nit: To check the specific error message
```suggestion
        replacement_parent_tx = self.wallet.create_self_transfer(
            from_node=self.nodes[0],
            utxo_to_spend=confirmed_utxos[0],
            fee_rate=Decimal('0.01'),
            mempool_valid=False
        )['hex']
        assert_raises_rpc_error(-26, 'too many potential replacements', self.nodes[0].sendrawtransaction, replacement_parent_tx, 0)
```